client/src/main/java/com/tigrisdata/db/client/service/StandardTigrisCollection.java

```diff
@@ -50,44 +52,28 @@
   private final String collectionName;
   private final Class<T> collectionTypeClass;
   private final TigrisDBGrpc.TigrisDBBlockingStub stub;
-  private final ObjectMapper objectMapper;
 
   StandardTigrisCollection(
       String databaseName, Class<T> collectionTypeClass, TigrisDBGrpc.TigrisDBBlockingStub stub) {
     this.databaseName = databaseName;
     this.collectionName = collectionTypeClass.getSimpleName().toLowerCase();
     this.collectionTypeClass = collectionTypeClass;
     this.stub = stub;
-    this.objectMapper = new ObjectMapper();
   }
 
   @Override
   public Iterator<T> read(
       TigrisFilter filter, ReadFields fields, ReadRequestOptions readRequestOptions)
       throws TigrisDBException {
     try {
-      Api.ReadRequestOptions readRequestOptionsAPI =
-          Api.ReadRequestOptions.newBuilder()
-              .setLimit(readRequestOptions.getLimit())
-              .setSkip(readRequestOptions.getSkip())
-              .build();
-
-      Api.ReadRequest.Builder readRequestBuilder =
-          Api.ReadRequest.newBuilder()
-              .setDb(databaseName)
-              .setCollection(collectionName)
-              .setFilter(ByteString.copyFrom(filter.toJSON(), StandardCharsets.UTF_8))
-              .setOptions(readRequestOptionsAPI);
-      if (!fields.isEmpty()) {
-        readRequestBuilder.setFields(ByteString.copyFromUtf8(fields.toJSON()));
-      }
-
-      Iterator<Api.ReadResponse> readResponseIterator = stub.read(readRequestBuilder.build());
+      Api.ReadRequest readRequest =
+          toReadRequest(databaseName, collectionName, filter, fields, readRequestOptions);
+      Iterator<Api.ReadResponse> readResponseIterator = stub.read(readRequest);
 
       Function<Api.ReadResponse, T> converter =
           readResponse -> {
             try {
-              return objectMapper.readValue(
+              return Utilities.OBJECT_MAPPER.readValue(
```

Comment:
I'd avoid referencing a static instance and prefer passing in `objectMapper` to the constructor for future flexibility. By default they can all be passed the same instance for the same effect.

Reply:
removed the static reference and added ability to pass in. I will revisit the visibility of some of these methods (For example toJSON() on filter shouldn't be public facing.

Reply:
Thanks @mattayres